src/north/amazon/AmazonS3.class.spec.js

```diff
@@ -0,0 +1,50 @@
+const fs = require('fs')
+const path = require('path')
+const AWS = require('aws-sdk')
+const ApiHandler = require('../ApiHandler.class')
+const AmazonS3 = require('./AmazonS3.class')
+const config = require('../../../tests/testConfig').default
+
+// Mock database service
+jest.mock('../../services/database.service', () => {})
+
+// Mock logger
+jest.mock('../../engine/Logger.class')
+
+// Mock engine
+const engine = jest.genMockFromModule('../../engine/Engine.class')
+engine.configService = { getConfig: () => ({ engineConfig: config.engine }) }
+
+beforeEach(() => {
+  jest.resetAllMocks()
+})
+
+describe('Amazone S3 north', () => {
+  const AmazonS3Config = config.north.applications[8]
+  const AmazonS3North = new AmazonS3(AmazonS3Config, engine)
+
+  it('should be properly initialized', () => {
+    expect(AmazonS3North.bucket).toEqual(AmazonS3Config.AmazonS3.bucket)
+    expect(AmazonS3North.folder).toEqual(AmazonS3Config.AmazonS3.folder)
+    AWS.config.update = jest.fn()
+    const accessKeyId = '' // doubt about what to put here
+    const secretAccessKey = ''
+    expect(AWS.config.update).toHaveBeenCalledWith(accessKeyId, secretAccessKey)
```

Comment:
It should have been called with the id you put in the config. So you don't need to create new variables and you can directly check if the method has been called with AmazonS3Config.AmazonS3(accessKeyId, secretKey)

Be careful, the secret key must be decrypted so it is not the string stored in the config. You must used the return value of the mocked method of the encryption service